grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -213,16 +213,24 @@ class InvoiceService {
     def updateItems(Invoice invoice, List items) {
         items.each { item ->
             InvoiceItem invoiceItem = InvoiceItem.get(item.id)
```

Comment:
Maybe it will be better to use `getAll` of this invoiceItem before starting iterating over this collection. For now, you are making calls to the database every iteration.

Reply:
I see your point, but then this would require a bigger refactor of this whole updateItems method because we have two cases we need to cover
if invoiceItem exists then update it
if it doesn't;t then create a new one out of invoice item candidate.

This might be worth looking into so thanks for pointing this out, but I am going to have to think about it and find a better, more optimal approach.